src/abstractdataframe/abstractdataframe.jl

```diff
@@ -970,15 +970,52 @@ julia> dropmissing(df, [:x, :y])
 @inline function dropmissing(df::AbstractDataFrame,
                              cols::Union{ColumnIndex, MultiColumnIndex}=:;
                              view::Bool=false, disallowmissing::Bool=!view)
+    # Identify Bool mask of which rows have no Missings
     rowidxs = completecases(df, cols)
     if view
         if disallowmissing
             throw(ArgumentError("disallowmissing=true is incompatible with view=true"))
         end
         return Base.view(df, rowidxs, :)
     else
-        newdf = df[rowidxs, :]
-        disallowmissing && disallowmissing!(newdf, cols)
+        # With exact indices we can skip a lot of iterations (as opposed to indexing via a mask)
+        selected_rows = _findall(rowidxs)
+        new_columns = Vector{AbstractVector}(undef, ncol(df))
+        df_columns = if typeof(df) == DataFrame 
+            _columns(df)
+        else 
+            # for SubDataFrame
+            collect(eachcol(df))
+        end
```

Comment:
```suggestion
        df_columns = df isa DataFrame ? _columns(df) : collect(AbstractVector, eachcol(df))
```
(`AbstractVector` to reduce compilation, but it probably is a minor thing)

Reply:
Updated!